### ISSUE TITLE
Cross-ref between Suricata alerts and related Zeek context events

### ISSUE THREAD

**Issue by philrz** [brimsec/brim#1250]
In the next Brim release, we'll have a field `community_id` present/populated in all Zeek events and Suricata alerts generated from pcaps that users import. These create a logical point to bi-directionally link these two data sources together.

For example, using the `all.pcap` test data, let's say I found my way to this Suricata alert.

![image](https://user-images.githubusercontent.com/5934157/100921535-8c14ae80-3491-11eb-95cf-7d3928f25811.png)

Today, by right-clicking on the `community_id` value and selecting "New search with this value", my results show me both the Suricata alert and any other records that share that value, which might include other Suricata alerts as well as Zeek `conn` events that were all generated from the same flow.

![image](https://user-images.githubusercontent.com/5934157/100921727-cc742c80-3491-11eb-9598-1c7165ff7e8b.png)

However, we could improve on this by having the Log Detail view automatically show the records returned from a query `community_id` query. This would show both the Suricata+Zeek events together similar to what we already do for the "uid correlation" view in Zeek events.

Note that on the Zeek side, only the `conn` event has the `community_id` field. Therefore, to make this most useful we'd ideally want to include in the same visualization all the Zeek events that share the same `uid` as the Zeek `conn` event that had the matching `community_id`. For example, following on what I showed above, if I right-click on the `uid` field in that `conn` record, I see:

![image](https://user-images.githubusercontent.com/5934157/100922311-97b4a500-3492-11eb-9292-9f954af697c5.png)

In other words, something I could have been shown in the first Log Detail view for the Suricata alert could effectively be a top-down waterfall of all these records:

![image](https://user-images.githubusercontent.com/5934157/100922556-e2ceb800-3492-11eb-934a-3ada0049a99d.png)

As we discussed recently as a team, this "chaining" of queries is something we already do in the app, i.e. when a user clicks on an app-level Zeek event like `dns` or `http`, the app automatically triggers a query for the `uid` of that event to find the corresponding `conn` event, since if we find it we know we have the 5-tuple and time information to extract a flow from the pcap, and hence we activate the "Packets" button.

Whatever visualization we add here should be "bi-directional". That is, above I described clicking on a Suricata alert and seeing the waterfall that includes all the Zeek events, but we'd want to make sure if a user clicks on any Zeek event, the same type of triggered query shows them any corresponding Suricata alerts. This will be essential to allow users to quickly pivot in Log Details between the alerts and the context about why they fired.

Assuming we end up with a single united view that shows both Zeek+Suricata events in the same waterfall and it takes the place of what we today call "uid correlation", it seems a name change is in order. Perhaps just call it "Correlations"? I'm open to suggestions here (cc: @orochford).

**Comment by philrz** [comment]
#1262 covers us in the Zeek->Suricata direction. A PR is still on the way for the Suricata->Zeek direction.

**Comment by philrz** [comment]
I've verified the Zeek->Suricata cross ref committed in #1262 on Brim commit `6a5cf3c` which talks to `zqd` commit `8c60a42d`.

Per the attached video, now when looking at any Zeek events in Log Detail, in addition to the Zeek events at many layers that have always been joined in the visualization via Zeek `uid`, now if there's also a Suricata alert that shares the same `community_id` as the Zeek `conn` record, that's also shown in the waterfall with an `alert` tile. That alert can then be clicked to pivot to it and see details about it. The Packets button is also activated when looking at the alert, such that when clicked the flow referenced by 5-tuple and `ts`+`duration` in the linked `conn` record is pulled up for packet-level analysis, if desired.

[Verify.zip](https://github.com/brimsec/brim/files/5691384/Verify.zip)

We've observed a strange symptom where sometimes Suricata seems to generate alerts with timestamps that are beyond the end of what we believe to be the end of the connection based on the `ts`+`duration` from the Zeek `conn` record. For now there's a fudge factor of 90 seconds added to the ending time range for the query that joins records on `community_id` to maximize our chance of including the future-timestamped alerts in the visualization. We hope to make adjustments to this logic as we continue to better understand this via closer study of Suricata.
